lib/index.js

```diff
@@ -18,75 +18,17 @@
+
 // Prevent React Native packager from seeing modules required with this
 const nodeRequire = require;
 
-function getContext() {
-    // If process.release.name is an object with we're probably running in Node or Electron
-    // From: http://stackoverflow.com/a/24279593/1417293
-    if (typeof process === 'object' && typeof process.release === "object" && process.release.name === "node") {
-
-        // Visual Studio Code defines the global.__debug__ object.
-        if (typeof global === 'object' && global.__debug__) {
-            return 'vscodedebugger';
-        }
-
-        return typeof process.versions === "object" && typeof process.versions.electron === "string" ? 'electron' : 'node.js';
-    }
-
-    // When running via Jest, the jest object is defined.
-    if (typeof jest === 'object') {
-        return 'node.js';
-    }
-
-    if (typeof navigator === 'object' && navigator.product === 'ReactNative') { // eslint-disable-line no-undef
-        // Visual Studio Code defines the global.__debug__ object.
-        if (typeof global !== 'undefined' && global.__debug__) {
-            return 'vscodedebugger'
-        }
-
-        // Check if its in remote js debugging mode
-        // https://stackoverflow.com/a/50377644
-        if (typeof DedicatedWorkerGlobalScope !== 'undefined') {
-            return 'chromedebugger';
-        }
-
-        // Otherwise, we must be in a "normal" react native situation.
-        // In that case, the Realm type should have been injected by the native code.
-        // If it hasn't, the user likely forgot to install the RealmJS CocoaPod
-        if (typeof Realm === 'undefined') {
-            throw new Error('Missing Realm constructor. Did you run "pod install"? Please see https://realm.io/docs/react-native/latest/#missing-realm-constructor for troubleshooting');
-        }
-
-        return 'reactnative';
-    }
-
-    // If we're not running in React Native but we already injected the Realm class,
-    // we are probably running in a pure jscore environment
-    if (typeof Realm !== 'undefined') {
-        return 'jscore';
-    }
-
-    // Visual Studio Code defines the global.__debug__ object.
-    if (typeof global !== 'undefined' && global.__debug__) {
-        return 'vscodedebugger';
-    }
-
-    // Finally, if the navigator.userAgent contains the string "Chrome", we're likely
-    // running via the chrome debugger, even if navigator.product isn't set to "ReactNative"
-    if (typeof navigator !== 'undefined' &&
-        /Chrome/.test(navigator.userAgent)) { // eslint-disable-line no-undef
-        return 'chromedebugger';
-    }
-
-    throw new Error("Unknown execution context");
-}
-
 function getRealmConstructor(context) {
     switch(context) {
         case 'node.js':
         case 'electron':
-            nodeRequire('./submit-analytics')('Run', context);
+            var analytics = nodeRequire('./submit-analytics');
+            analytics.submitProductionAnalytics('Run');
```

Comment:
We only submit analytics for node/electron and not React Native?

Reply:
As per extensive discussions, it will be kept the way it is for now.
Possible reimplementation to be discussed in the future.